ister.py

```diff
@@ -78,9 +79,11 @@ def run_command(cmd, raise_exception=True, log_output=True, environ=None):
         for line in lines:
             decoded_line = line.decode('ascii', 'ignore').rstrip()
             output.append(decoded_line)
-            if log_output:
+            if show_output:
+                LOG.info(decoded_line)
+            elif log_output:
                 LOG.debug(decoded_line)
-        if proc.wait() != 0 and raise_exception:
```

Comment:
So this is interesting to me because I expected it to cause issues without the threading but when I looked closer at the gui code doing this I couldn't understand why. The reason this works (and why poll() vs wait() is actually a non issue here) is because the iterator is doing the work of reading through the pipe and will only terminate in this case once the pipe closes. Having this be poll is nicer than doing the comparison though I guess so I'll take it as is.